Modified challenge create page

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->


#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Changes made in UI to make it look similar to current Eval AI application.
- Changes made in challenge service to subscribe for api callback of createChallenge method.
- Changes made in the logic to handle the file upload and show related errors and toast.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-150-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
![Screenshot from 2019-06-16 10-49-36](https://user-images.githubusercontent.com/17106489/59566010-0b0cf180-9028-11e9-8790-246b4a60b7ec.png)

![Screenshot from 2019-06-16 10-49-54](https://user-images.githubusercontent.com/17106489/59566014-1233ff80-9028-11e9-9c73-94c7cce96d5b.png)

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Modified footer component

Added github fork and star button using github api

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Changes in app component

Modified app component to show loader according to authService 'isLoader' variable.

sanketbansal committed:
Modified app loader component

app loader component modified to accompany AngularJS application UI.

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Added side bar component

Added side bar as utility component for the side bar in the Dashboard UI

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

sanketbansal committed:
changes in style files

Added font variables in ported variable file
Added font style in ported main file.

sanketbansal committed:
Modified footer

Made the footer UI similar to the angularjs application one.

sanketbansal committed:
Added simple header component

simple header added to modify the UI of the header in the dashboard

sanketbansal committed:
Modifications in simple header component

Added color property for signup button
Added styles for login button

sanketbansal committed:
Modified app component

Removed ngIf  condition from app-loading component to resolve the error:
ExpressionChangedAfterItHasBeenCheckedError

sanketbansal committed:
Modified challenge create component

Changes made in UI to make it look similar to current Eval AI application
Changes made in challenge service to subscribe for api callback of createChallenge method.
Changes made in the logic to handle the file upload and show related errors and toast.

sanketbansal commented:
@lunayach @Shekharrajak please review the PR.

Shekharrajak commented:
Similar commits -- https://github.com/Cloud-CV/EvalAI-ngx/pull/149#issuecomment-503590467

sanketbansal committed:
Added feature to start stop loader

Added methods which are used by the components to start and stop the loader.

sanketbansal committed:
Moved styles in challenge-create component

Styles related to challenge-create component moved from 'styles-port/modules/challenge-create.scss'

sanketbansal committed:
Fixed lint error in angular.json file

Added comma after a property in angular.json file

sanketbansal committed:
Added basic styles

Basic style files added in styles-port folder

sanketbansal committed:
Removed duplicate styles

Removed duplicate scss files from basic.scss

sanketbansal committed:
Removed unused css from main.scss

Duplicates removed from main.scss file

sanketbansal committed:
Removed unused assets and styles folder

Removed unused images-port folder and styles-port folder

sanketbansal committed:
Removed unused and duplicate code

Removed unused methods and variable
Duplicat side bar component inpoer removed from

sanketbansal committed:
Added style files

Added basic and main scss file in styles-port folder

sanketbansal committed:
Removed unused css from main.scss

Duplicates removed from main.scss file

sanketbansal committed:
removed basic.scss file

Removed basic.scss file from styles-folder

sanketbansal committed:
Removed logs

Removed comments and logs

sanketbansal committed:
Removed main.scss file

Removed main.scss file from styles-folder

sanketbansal committed:
Added component specific styles

component styles from main.scss file

sanketbansal committed:
changes made in html files

Removed html-styles from simple header component
Renamed styles class name in challenge create component

sanketbansal committed:
Modified app loader component

app loader component modified to accompany AngularJS application UI.

sanketbansal committed:
Added styles for loading component

Added styles in base.scsss and variable.scss reqiored by loading component
Added styles in loading.component.scss file

sanketbansal committed:
Added loader component

Added modifed loader styles and html

sanketbansal committed:
Removed comments

Commented lines removed from simple header file

Shekharrajak commented:
Still 37 file changes. I can't review it. Also, I see changes like other PRs.

sanketbansal commented:
> Still 37 file changes. I can't review it. Also, I see changes like other PRs.

The file changes will get further reduce after merging #141 . I kept these changes as these are required for this PR also.

Shekharrajak commented:
Okay thanks for letting me know.

sanketbansal committed:
changes in challenge-create page

Removed unused styles and component folder.
Removed console logs and comments.

sanketbansal commented:
@lunayach this can be reviewed also.

lunayach committed:
Merge branch 'master' into modified-challengeCreate

lunayach commented:
@sanketbansal can you confirm if you are able to create a host team on your end. When I navigate to `Create Challenge`, it redirects me to `/teams/hosts`. There, just an empty page appears.

sanketbansal commented:
> @sanketbansal can you confirm if you are able to create a host team on your end. When I navigate to `Create Challenge`, it redirects me to `/teams/hosts`. There, just an empty page appears.

I think that issue should be resolved after merging #149

lunayach committed:
Merge branch 'master' into modified-challengeCreate

lunayach commented on src/app/components/challenge-create/challenge-create.component.scss:
```diff
@@ -1,38 +1,72 @@
+    content: '';
+    position: relative;
+    display: inline-block;
+    color: black;
```
Remove this.

sanketbansal replied:
Done

lunayach commented on src/app/components/challenge-create/challenge-create.component.html:
```diff
@@ -1,27 +1,56 @@
   </div>
 </div>
-<app-footer></app-footer>
+<div class="clearfix"></div>
```
Is this needed?

sanketbansal replied:
Removed

lunayach commented on src/app/components/challenge-create/challenge-create.component.ts:
```diff
@@ -73,33 +83,51 @@ export class ChallengeCreateComponent implements OnInit {
+        );
+    } else {
+      this.isFormError = true;
+      this.globalService.showToast('error', 'Please Upload File');
```
The position of this is not aligned correctly. Render it directly below the `Upload file` text field.

lunayach commented on src/app/components/challenge-create/challenge-create.component.html:
```diff
@@ -1,27 +1,56 @@
+                  <button class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-14" type="submit" value="Submit">Submit</button>
+                </div>
+
+                <div *ngIf="isFormError" class="wrn-msg text-highlight align-right"> Please Upload File </div>
```
The position of this is not aligned correctly. Render it directly below the `Upload file` text field.

sanketbansal replied:
Done

lunayach reviewed (changes requested):
The functionality looks good. Minor changes needed.

sanketbansal committed:
Changes in challenge-create component

Aligned form error with text field

lunayach reviewed (approved):
@RishabhJain2018, Looks good to be merged.
